app/src/main/java/ca/marshallasch/veil/services/VeilService.java

```diff
@@ -0,0 +1,143 @@
+     */
+    private final class ServiceHandler extends Handler {
+        public ServiceHandler() {
+
+        }
+
+        /**
+         * function for creating the ServiceHandler
+         * @param looper thread's looper
+         */
+        public ServiceHandler(Looper looper){
+            super(looper);
+        }
+
+        /**
+         * Where messages to do work on the thread is processed
+         * @param msg work message
+         */
+        @Override
+        public void handleMessage(Message msg) {
+            switch (msg.what){
+                case ACTION_VIEW_MESH_SETTINGS:
+                    rightMeshController.showMeshSettings();
+                    break;
+                case ACTION_MAIN_RESUME_MESH:
+                    rightMeshController.resumeMeshManager();
+                    break;
+                case ACTION_MAIN_REFRESH_PEER_LIST:
+                    rightMeshController.getPeers();
+                    break;
+                case ACTION_MAIN_MANUAL_REFRESH:
+                    rightMeshController.manualRefresh();
+                    break;
+                default:
+                    super.handleMessage(msg);
+            }
+        }
+    }
+
+    /**
+     * Target for clients to send messages to ServiceHandler
+     */
+    final Messenger veilMessenger = new Messenger(new ServiceHandler());
+
+
+    /**
+     * Connects to RightMesh when service is started
+     */
+    @Override
+    public void onCreate(){
+        super.onCreate();
+
+        //Start up thread and make it background priority so it doesn't disrupt UI
+        HandlerThread veilServiceThread = new HandlerThread("VeilServiceThread",
+                Process.THREAD_PRIORITY_BACKGROUND);
+        veilServiceThread.start();
+
+        //Get the handlerThread's Looper and use it for handler
+        veilServiceLooper = veilServiceThread.getLooper();
+        veilServiceHandler = new ServiceHandler(veilServiceLooper);
+
+
+        //start RightMesh connection through controller using service context
+        rightMeshController = new RightMeshController();
+        rightMeshController.connect(this);
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        Toast.makeText(this, "Veil service started.", Toast.LENGTH_SHORT).show();
+
+        Message msg = veilServiceHandler.obtainMessage();
+        msg.arg1 = startId;
+        veilServiceHandler.sendMessage(msg);
+
+        return START_STICKY;
+    }
+
+    /**
+     * Disconnects from RightMesh when service is stopped
+     */
+    @Override
+    public void onDestroy(){
+        super.onDestroy();
+        Toast.makeText(this, "Veil service ended.", Toast.LENGTH_SHORT).show();
+        rightMeshController.disconnect();
+        rightMeshController = null;
+        //force kill service so it has no chance of recreating itself
+        android.os.Process.killProcess(android.os.Process.myPid());
```

Comment:
If you kill the process will that kill the app as well? since the service is int eh same process as the activity for the app?  Is this good practice?

Reply:
Killing the service won't kill the app at the moment. I feel like it should though since there is a risk for a crash if the app makes a call to the service but it stop running.

Reply:
Right now killing the service does not kill the app but vice versa it does. I feel we should change to if the service ends so should the app. Removed the potential of the app making a request without the service existing causing a crash. Your thoughts?